File: 0070-climbing-stairs/0070-climbing-stairs.java
class Solution {
    public int climbStairs(int n) {
        if (n <= 2){
            return n;
        }
        int p2 = 1;
        int p1 = 2;
        for(int i = 3; i <= n; i++){
            int curr = p1 + p2;
            p2 = p1;
            p1 = curr;
        }
        return p1;
    }
}

File: 0070-climbing-stairs/0070-climbing-stairs.py
class Solution:
    def climbStairs(self, n: int) -> int:
        if n <= 2:
            return n
        p2 = 1
        p1 = 2
        for i in range(3, n+1):
            curr = p1 + p2
            p1, p2 = curr, p1
        return p1